client/chat/channel-info-card.tsx

```diff
@@ -92,7 +135,7 @@ export function ConnectedChannelInfoCard({
     setIsJoinInProgress(true)
     dispatch(
       joinChannel(channelName, {
-        onSuccess: channel => navigateToChannel(channel.id, channel.name),
+        onSuccess: () => {},
         onError: err => {
           setJoinChannelError(err)
```

Comment:
You need to do something with this, putting it in state and then not looking at that state isn't helping anybody (but again, we have no guarantees that this component is even still mounted when this gets called, so using the card as a surface for displaying any error is a bad idea)

Reply:
I am looking at this, but the only thing I'm using it in this card is to disable the "Join" button if you clicked on it and you get an error that you're banned (the snackbar displays the actual error from action-creators).

I initially thought to just close the popover whenever you click the "Join" button, but this component doesn't assume it's rendered in a Popover (and it actually isn't when it's rendered on a channel page).

Reply:
I mean that is probably an indication that it should be passed a (perhaps optional) event handler and you can do that. At the very least, I will again state that you need to do something with this error. You should not really assume it is the singular error you "expect". If all types of errors are handled elsewhere that's fine, but in that case the proper thing to do would be to note that down here, and set boolean state for whether to disable the button. What you've written here gives the impression that the error will be handled by something when you're just using it as a flag, and that's dangerous if people come along and change the place you *expect* it to have been handled.